async_service/asyncio.py

```diff
@@ -118,6 +118,17 @@ async def _handle_cancelled(self) -> None:
         When cancellation is requested this triggers the cancellation of all
         background tasks.
         """
+        try:
+            await self._real_handle_cancelled()
+        except asyncio.CancelledError:
+            raise
+        except Exception:
```

Comment:
@pipermerriam Since we re-raise the exception here, I'm going to change this to `BaseException` to make sure we log all possible exceptions